### PR TITLE
fix(#11545): data Insight report Improvements

### LINKED ISSUE #11545: Data Insight Report Improvements
- Need a Send Now Button on the Data Insight Report Page 
- Add an Email field to teams while creating the team.

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes:

Fixes #11545 

I worked on issue #11545 

image

https://github.com/open-metadata/OpenMetadata/assets/59080942/99fae883-a001-43ca-85e3-6600f986018b

#
### Type of change:

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Review by mohityadav766 — APPROVED** [review]

File: openmetadata-ui/src/main/resources/ui/src/pages/AlertDataInsightReportPage/AlertDataInsightReportPage.tsx [hunk 1]
```diff
@@ -65,6 +67,18 @@ const AlertDataInsightReportPage = () => {
     }
   }, []);
 
+  const handleSendDataInsightReport = async () => {
+    try {
+      setIsSendingReport(true);
+      await triggerEventById(dataInsightAlert?.id ?? '');
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/AlertDataInsightReportPage/AlertDataInsightReportPage.tsx:73]: If we don't have id with us we can return from above only right?

File: openmetadata-ui/src/main/resources/ui/src/pages/AlertDataInsightReportPage/AlertDataInsightReportPage.tsx [hunk 2]
```diff
@@ -102,14 +116,22 @@ const AlertDataInsightReportPage = () => {
...
+              <Button icon={<Icon component={IconEdit} size={12} />}>
+                {t('label.edit')}
+              </Button>
+            </Link>
+            <Button
+              data-testid="send-now-button"
+              icon={<Icon component={IconSend} size={12} />}
+              loading={isSendingReport}
+              onClick={handleSendDataInsightReport}>
+              {t('label.send-now')}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/AlertDataInsightReportPage/AlertDataInsightReportPage.tsx:132]: What if there's no `dataInsightAlert` or there will be always one?
Also styling for button is not same as what we have, Can we loose icon here? Looks unwanted

File: openmetadata-ui/src/main/resources/ui/src/pages/teams/AddTeamForm.tsx [hunk 3]
```diff
@@ -157,6 +157,17 @@ const AddTeamForm: React.FC<AddTeamFormType> = ({
...
+          label={t('label.email')}
+          name="email"
+          rules={[{ type: 'email' }]}>
+          <Input
+            data-testid="email"
+            placeholder={t('label.enter-entity', {
+              entity: t('label.email-lowercase'),
+            })}
+          />
+        </Form.Item>
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/teams/AddTeamForm.tsx:170]: Can we update the Cypress too?

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/pages/teams/AddTeamForm.tsx:170]: Sure.

File: openmetadata-ui/src/main/resources/ui/src/pages/AlertDataInsightReportPage/AlertDataInsightReportPage.tsx (hunk 2 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/pages/AlertDataInsightReportPage/AlertDataInsightReportPage.tsx:132]: We have system-generated `dataInsightAlert`  so there will always be one.
Button styling is the same as we have on other alert pages.

File: openmetadata-ui/src/main/resources/ui/src/locale/languages/en-us.json [hunk 4]
```diff
@@ -1027,6 +1028,8 @@
     "data-insight-message": "Manage data insights pipeline.",
     "data-insight-page-views": "Displays the number of times a dataset type was viewed.",
     "data-insight-pipeline-description": "Monitor data usage and set up KPIs by deploying the data insights pipeline. Refer to our documentation to learn more <0>{{link}}</0>",
+    "data-insight-report-send-failed-message": "Failed to send data insight report!",
+    "data-insight-report-send-success-message": "Data Insight Report sent successfully!",
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/locale/languages/en-us.json:1032]: Let's remove "!" 
cc: @ShilpaVernekar

File: openmetadata-ui/src/main/resources/ui/src/pages/AlertDataInsightReportPage/AlertDataInsightReportPage.tsx [hunk 5]
```diff
@@ -65,6 +67,20 @@ const AlertDataInsightReportPage = () => {
...
+      try {
+        setIsSendingReport(true);
+        await triggerEventById(dataInsightAlert.id);
+        showSuccessToast(t('message.data-insight-report-send-success-message'));
+      } catch (error) {
+        showErrorToast(t('message.data-insight-report-send-failed-message'));
+      } finally {
+        setIsSendingReport(false);
+      }
+    }
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/pages/AlertDataInsightReportPage/AlertDataInsightReportPage.tsx:81]:
```suggestion

    if (isUndefined(dataInsightAlert)) {
      return;
    }
      try {
        setIsSendingReport(true);
        await triggerEventById(dataInsightAlert.id);
        showSuccessToast(t('message.data-insight-report-send-success-message'));
      } catch (error) {
        showErrorToast(t('message.data-insight-report-send-failed-message'));
      } finally {
        setIsSendingReport(false);
      }
    
```

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/pages/AlertDataInsightReportPage/AlertDataInsightReportPage.tsx:81]: We are passing on the handler to button onClick and it expected return type void if we follow the above approach then the sonar cloud will flag it as a bug.

File: openmetadata-ui/src/main/resources/ui/src/locale/languages/en-us.json (hunk 4 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/locale/languages/en-us.json:1032]: Sure, can we take care of this in another patch?

Comment (ShilpaVernekar) [reply openmetadata-ui/src/main/resources/ui/src/locale/languages/en-us.json]: @chirag-madlani Yeah doesn't make sense to use "!" for a failed to send report. But to just add a full stop.

**Review by chirag-madlani — APPROVED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Comment by harshach** [comment]
@Sachin-chaurasiya @mohityadav766 can we add a preview of the email about to send out in the UI?

**Review by harshach — APPROVED** [review]

**Comment by mohityadav766** [comment]
> @Sachin-chaurasiya @mohityadav766 can we add a preview of the email about to send out in the UI?

@harshach that preview is at org level / or team level , since org level looks same , but team level email differs per team?

**Pull request merged**